packages/component/src/hooks/useEmojiFromStyles.js

```diff
@@ -0,0 +1,51 @@
+/*eslint no-useless-escape: "off"*/
+/*eslint require-unicode-regexp: "off" */
+import useStyleOptions from './useStyleOptions';
+
+function escapeRegexp(emoticon) {
+  return emoticon.replace(/[\\^$*+?.()|[\]{}]/g, '\\$&');
+}
+
+export default function useEmojiFromStyles() {
+  const [{ emojiAutocorrect: autocorrect, emojiList: customEmojiList }] = useStyleOptions();
+
+  const emojiAutocorrect = autocorrect || false;
+
+  const emojiUnicodeMap = customEmojiList || {
+    ':)': '😊',
+    ':-)': '😊',
+    '(:': '😊',
+    '(-:': '😊',
+    ':-|': '😐',
+    ':|': '😐',
+    ':-D': '😀',
+    ':D': '😀',
+    ':-p': '😛',
+    ':p': '😛',
+    ':-P': '😛',
+    ':P': '😛',
+    ':-o': '😲',
+    ':o': '😲',
+    ':O': '😲',
+    ':-O': '😲',
+    ':-0': '😲',
+    ':0': '😲',
+    ';-)': '😉',
+    ';)': '😉',
+    '<3': '❤️',
+    '</3': '💔',
+    '<\\3': '💔'
+  };
+
+  const escapedString =
```

Comment:
I like this solution. Would it make more sense not to expose emojiRegExp at all and just do this logic inside useRenderEmoji instead? That way you wouldn't need to pass it between the two, it would just always be correct in the one place it needs to be.